DQM/L1TMonitor/src/L1TStage2BMTF.cc

```diff
@@ -35,70 +36,81 @@ void L1TStage2BMTF::beginLuminosityBlock(const edm::LuminosityBlock& iLumi, cons
   bmtf_hwPt_hwEta->setTitle(";HW p_{T}; HW #eta");
 
-  bmtf_hwPt_hwLocalPhi  = ibooker.book2D("bmtf_hwPt_hwLocalPhi" , "HW p_{T} vs HW Local #phi", 511, -0.5, 510.5, 201, -100.5, 100.5);
+  bmtf_hwPt_hwLocalPhi  = ibooker.book2D(histoPrefix+"_hwPt_hwLocalPhi" , "HW p_{T} vs HW Local #phi", 511, -0.5, 510.5, 201, -100.5, 100.5);
   bmtf_hwPt_hwLocalPhi->setTitle(";HW p_{T}; HW Local #phi");
 
-  bmtf_hwEta_bx    = ibooker.book2D("bmtf_hwEta_bx"   , "HW #eta vs BX"      , 461, -230.5, 230.5,  5, -2.5, 2.5);
+  bmtf_hwEta_bx    = ibooker.book2D(histoPrefix+"_hwEta_bx"   , "HW #eta vs BX"      , 461, -230.5, 230.5,  5, -2.5, 2.5);
   bmtf_hwEta_bx->setTitle(";HW #eta; BX");
 
-  bmtf_hwLocalPhi_bx    = ibooker.book2D("bmtf_hwLocalPhi_bx"   , "HW Local #phi vs BX"      , 201, -100.5, 100.5,  5, -2.5, 2.5);
+  bmtf_hwLocalPhi_bx    = ibooker.book2D(histoPrefix+"_hwLocalPhi_bx"   , "HW Local #phi vs BX"      , 201, -100.5, 100.5,  5, -2.5, 2.5);
   bmtf_hwLocalPhi_bx->setTitle(";HW Local #phi; BX");
 
-  bmtf_hwPt_bx     = ibooker.book2D("bmtf_hwPt_bx"    , "HW p_{T} vs BX"     , 511,   -0.5, 510.5,  5, -2.5, 2.5);
+  bmtf_hwPt_bx     = ibooker.book2D(histoPrefix+"_hwPt_bx"    , "HW p_{T} vs BX"     , 511,   -0.5, 510.5,  5, -2.5, 2.5);
   bmtf_hwPt_bx->setTitle(";HW p_{T}; BX");
 
-  bmtf_hwQual_bx   = ibooker.book2D("bmtf_hwQual_bx"  , "HW Quality vs BX"      ,  20,   -0.5,  19.5,  5, -2.5, 2.5);
+  bmtf_hwQual_bx   = ibooker.book2D(histoPrefix+"_hwQual_bx"  , "HW Quality vs BX"      ,  20,   -0.5,  19.5,  5, -2.5, 2.5);
   bmtf_hwQual_bx->setTitle("; HW Quality; BX");
-
-  // bmtf_twinmuxInput_PhiBX = ibooker.book1D("bmtf_twinmuxInput_PhiBX"  , "TwinMux Input Phi BX"      ,  5, -2.5, 2.5);
-  // bmtf_twinmuxInput_PhiPhi = ibooker.book1D("bmtf_twinmuxInput_PhiPhi"  , "TwinMux Input Phi HW Phi"      , 201, -100.5, 100.5);
-  // bmtf_twinmuxInput_PhiPhiB = ibooker.book1D("bmtf_twinmuxInput_PhiPhiB"  , "TwinMux Input Phi HW PhiB"   , 201, -100.5, 100.5);
-  // bmtf_twinmuxInput_PhiQual = ibooker.book1D("bmtf_twinmuxInput_PhiQual"  , "TwinMux Input Phi HW Quality" , 20,   -0.5,  19.5);
-  // bmtf_twinmuxInput_PhiStation = ibooker.book1D("bmtf_twinmuxInput_PhiStation"  , "TwinMux Input Phi HW Station", 6, -1, 5);
-  // bmtf_twinmuxInput_PhiSector = ibooker.book1D("bmtf_twinmuxInput_PhiSector" , "TwinMux Input Phi HW Sector"    , 14, -1,  13 );
-  // bmtf_twinmuxInput_PhiWheel = ibooker.book1D("bmtf_twinmuxInput_PhiWheel"  , "TwinMux Input Phi HW Wheel"      , 16 , -4, 4);
-  // bmtf_twinmuxInput_PhiTrSeg = ibooker.book1D("bmtf_twinmuxInput_PhiTrSeg"  , "TwinMux Input Phi HW Track Segment"      , 6, -1, 5 );
-  // bmtf_twinmuxInput_PhiWheel_PhiSector = ibooker.book2D("bmtf_twinmuxInput_PhiWheel_PhiSector"  , "TwinMux Input Phi HW Wheel vs HW Sector", 16 , -4, 4, 14, -1,  13 );
+  if (kalman) {
+    bmtf_hwPt  = ibooker.book1D(histoPrefix+"_hwPt", "HW p_{T}", 522, -0.5, 521.5);
```

Comment:
hwPt and hwQual_bx MEs were already booked earlier. Do not book them again. If you want different axis ranges for Kalman use a variable to set them depending of Kalman is true or false.